sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -120,6 +137,8 @@
         ELSE statement_end_offset
     END - statement_start_offset) / 2) + 1) AS statement_text,
     SUBSTRING(qt.text, 1, {proc_char_limit}) as text,
+    OBJECT_NAME(sproc_object_id, s.dbid) as object_name,
+    OBJECT_SCHEMA_NAME(sproc_object_id, s.dbid) as schema_name,
```

Comment:
Have you perhaps tested agains SQL Server 2022 and adaptive plans, as the behaviour changed with the introduction of adaptive plans.

Reply:
Not yet but it's planned.

Reply:
We were able to verify that the procedure and query stats are accurate in 2022

Reply:
Actually, there is a serious mistake introduced with this change - all statistics will be `NULL` for the statements that were produced by parameter sensitive plan optimization (PSP). While it's generally not possible to associate PSP queries with stored procedures, the old query metrics query at least produces correct statistics for the individual queries. The demo is in the attached notebook:
[psp-sqlserver2022.pdf](https://github.com/DataDog/integrations-core/files/13249086/psp-sqlserver2022.pdf)

Reply:
https://datadoghq.atlassian.net/browse/DBMON-3089